### ISSUE TITLE
`Style/Alias` doesn't allow any use of `alias`

### ISSUE THREAD

**Issue by marxarelli** [bbatsov/rubocop#1460]
According to comments in bbatsov/ruby-style-guide#377 `alias` is not strictly forbidden by the style guide and in fact can be more appropriate than `alias_method` in some cases—usually those involving lexical scope and a clear indirection to methods within the same scope. The `Style/Alias` cop needs refactoring to _allow_ the use of `alias` in cases such as the following.

```ruby
class Westerner
  def first_name
    @names.first
  end

  alias given_name first_name
end
```

**Comment by bf4** [comment]
I see the PR is merged, but in `0.35.1` I'm still getting that annoying `Style/Alias: Use alias_method instead of alias. (https://github.com/bbatsov/ruby-style-guide#alias-method)` even though `https://github.com/bbatsov/ruby-style-guide#alias-method-lexically` it should be permitted.  And the only option is `Enabled`.  I don't really understand cop internals, so I can't tell what it's doing.

**Comment by bbatsov** [comment]
@marxarelli disappeared and I didn't have time to work on this myself. Someone else will have to drive the work here to completion. 

Basically, now the cop reflects an older version of the style guide.

**Comment by bf4** [comment]
I'm happy to help with some guidance.. I understand the ruby and the parser, but would need some help getting that into a tested coo

B mobile phone

**Comment by bbatsov** [comment]
Sure. You can start work on this and we'll lend you all the help you need.

**Comment by bf4** [comment]
@alexdowad Yay! :rainbow: :heart: :sparkles: :100:

**Comment by marxarelli** [comment]
Sorry about being a total flake on this one. I'm glad it got done by @alexdowad !

**Comment by matrinox** [comment]
I think the default should be prefer_alias_method. That was the behaviour before and upgrading a minor release should not break that expectation.

**Comment by bbatsov** [comment]
See https://github.com/bbatsov/rubocop#installation
